common-utils/src/main/java/eu/arrowhead/common/mqtt/MqttService.java

```diff
@@ -0,0 +1,144 @@
+import java.security.UnrecoverableKeyException;
+import java.security.cert.CertificateException;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.TrustManagerFactory;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.eclipse.paho.client.mqttv3.MqttClient;
+import org.eclipse.paho.client.mqttv3.MqttConnectOptions;
+import org.eclipse.paho.client.mqttv3.MqttException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Component;
+import org.springframework.util.Assert;
+
+import eu.arrowhead.common.Constants;
+import eu.arrowhead.common.SSLProperties;
+import eu.arrowhead.common.Utilities;
+
+@Component
+@ConditionalOnProperty(name = Constants.MQTT_API_ENABLED, matchIfMissing = false)
+public class MqttService {
+
+	//=================================================================================================
+	// members
+
+	private static final String SSL_KEY_MANAGER_FACTORY_ALGORITHM = "ssl.KeyManagerFactory.algorithm";
+	private static final String SSL_TRUST_MANAGER_FACTORY_ALGORITHM = "ssl.TrustManagerFactory.algorithm";
+	private static final String TLS_VERSION = "TLSv1.2";
+
+	@Autowired
+	private SSLProperties sslProperties;
+
+	private final Map<String, MqttClient> clientMap = new ConcurrentHashMap<>();
+
+	private final Logger logger = LogManager.getLogger(getClass());
+
+	//=================================================================================================
+	// methods
+
+	//-------------------------------------------------------------------------------------------------
+	public MqttClient client(final String connectId) {
+		Assert.isTrue(!Utilities.isEmpty(connectId), "connectId is empty");
+		return clientMap.get(connectId);
+	}
+
+	//-------------------------------------------------------------------------------------------------
+	public void connect(final String connectId, final String address, final int port, final String clientId, final String username, final String password) throws MqttException {
+		logger.debug("createConnection started");
+		Assert.isTrue(!Utilities.isEmpty(connectId), "connectId is empty");
+		Assert.isTrue(!Utilities.isEmpty(address), "address is empty");
+
+		if (clientMap.containsKey(connectId)) {
+			disconnect(connectId);
+		}
+
+		String serverURI = sslProperties.isSslEnabled() ? "ssl://" : "tcp://";
+		serverURI = serverURI + address + ":" + port;
+
+		final MqttConnectOptions options = new MqttConnectOptions();
+		options.setAutomaticReconnect(true);
+		options.setCleanSession(true);
+		if (!Utilities.isEmpty(username)) {
+			options.setUserName(username);
+		}
+		if (!Utilities.isEmpty(password)) {
```

Comment:
this if should go inside the previous if. Makes no sense to set a password if user is not set.